web3/contract.py

```diff
@@ -721,14 +741,23 @@ def __call__(self, *args, **kwargs):
             return super().__call__(*args, **kwargs)
 
 
+class NonExistentFallbackFunction(object):
```

Comment:
I'm struggling to find an alternate way to accomplish what this class does.

At first I was thinking of a simple `@property` method which raised the needed exception buts since we need to be able to access the `fallback` property on both the class and instance, that doesn't work.

That lead me to thinking about implementing this as a descriptor instead.

```python
class FallbackFunctionDescriptor:
    fallback_fn = None

    def __init__(self, fallback_fn=None):
        self.fallback_fn = None
    def __get__(self, instance, owner):
        ...
```

While it's still kind of messy, I think it's a little cleaner than raising the exception in `__getattr__` and it causes the exception to be raised at the moment that `c.fallback` is accessed, rather than my understanding of the current behavior which waits until an attribute on `c.fallback` is accessed.

Thoughts?

Reply:
Haven't thought much when I implemented `NonExistentFallbackFunction` :) so I don't have any preferences. It's up to you. 

I will implement it using `__get__` instead.